src/connectors/hits/__tests__/connectHits-test.ts

```diff
@@ -550,4 +550,165 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/hits/js/#co
       expect(nextState.highlightPostTag).toBe('</mark>');
     });
   });
+
+  describe('sendEvent & bindEvent', () => {
+    let instantSearchInstance;
+    let renderFn;
+    let hits;
+    beforeEach(() => {
+      renderFn = jest.fn();
+      const makeWidget = connectHits(renderFn);
+      const widget = makeWidget({});
+
+      const helper = algoliasearchHelper(createSearchClient(), '', {});
+      helper.search = jest.fn();
+
+      const initOptions = createInitOptions({
+        helper,
+        state: helper.state,
+      });
+      instantSearchInstance = initOptions.instantSearchInstance;
+      widget.init!(initOptions);
+
+      hits = [
+        {
+          objectID: '1',
+          fake: 'data',
+          __queryID: 'test-query-id',
+          __position: 0,
+        },
+        {
+          objectID: '2',
+          sample: 'infos',
+          __queryID: 'test-query-id',
+          __position: 1,
+        },
+      ];
+
+      const results = new SearchResults(helper.state, [
+        createSingleSearchResponse({ hits }),
+      ]);
+      widget.render!(
+        createRenderOptions({
+          results,
+          state: helper.state,
+          helper,
+        })
+      );
+    });
+
+    describe('sendEvent', () => {
+      it('sends view event when hits are rendered', () => {
```

Comment:
can you add a test that if some other widget causes a rerender, without changing search parameters, that it doesn't get called twice?

Reply:
@Haroenv Can you give me an idea? What kind of situation can it be?

Initially I thought of adding a `poweredBy` widget. But adding a new widget calls `scheduleSearch` in the end.

Is there a flow where InstantSearch doesn't trigger a search but just trigger a rerender of all widgets?

Reply:
for example adding a widget should indeed not cause a new view event, since the parameters are the same, but since we do a render (do we?), it could send the view event again.

Reply:
@Haroenv Oops. I missed this and already merged this PR. I can address it somewhere else.

> it could send the view event again.

Then we should somehow compare the search parameters and send view events only when they have changed. Is it similar to what you have in your mind?